application/actions/auth.go

```diff
@@ -146,60 +147,68 @@ func authInvite(c buffalo.Context) error {
 	return c.Render(http.StatusOK, render.JSON(resp))
 }
 
-func getOrgAndUserOrgs(
-	authEmail string,
-	c buffalo.Context) (models.Organization, models.UserOrganizations, error) {
+func getUserOrgs(c buffalo.Context, authEmail string) (models.UserOrganizations, error) {
 	var orgID int
 	oid := c.Param(OrgIDParam)
-	if oid == "" {
-		orgID = 0
-	} else {
+
+	if oid != "" {
 		orgID, _ = strconv.Atoi(oid)
 	}
 
-	var org models.Organization
 	var userOrgs models.UserOrganizations
 	if err := userOrgs.FindByAuthEmail(authEmail, orgID); err != nil {
-		return org, userOrgs, err
+		if domain.IsOtherThanNoRows(err) {
+			return userOrgs, err
+		}
 	}
 
-	if len(userOrgs) == 1 {
-		org = userOrgs[0].Organization
-	}
+	return userOrgs, nil
+}
 
-	// no user_organization records yet, see if we have an organization for user's email domain
-	if len(userOrgs) == 0 {
-		if err := org.FindByDomain(domain.EmailDomain(authEmail)); err != nil {
-			return org, userOrgs, err
-		}
-		if org.AuthType == "" {
-			return org, userOrgs, errors.New("unable to find organization by email domain")
-		}
+// no user_organization records yet, see if we have an organization for user's email domain
+func getOrgForNewUser(authEmail string) (models.Organization, error) {
+	var org models.Organization
+	if err := org.FindByDomain(domain.EmailDomain(authEmail)); err != nil {
+		return org, err
 	}
+	return org, nil
+}
 
-	return org, userOrgs, nil
+type authError struct {
+	httpStatus int
+	errorCode  string
+	errorMsg   string
 }
```

Comment:
It's a little complicated, because these are based on the `authRequestError` signature.
I think I'll leave `httpStatus` as is and change `errorCode` to `errorKey` here and in the `authRequestError` signature. 

Later, if we want, we can change the `authRequestError` signature to expect an `domain.AppError` param, instead of the `httpStatus` and `errorKey` params.